src/main/java/com/myThreadPool/RunnableDelayPairing.java

```diff
@@ -0,0 +1,27 @@
+package com.myThreadPool;
+
+/**
+ * This is a wrapper class to contain both the Runnable and its delay, after which it is supposed to execute
+ */
+public class RunnableDelayPairing {
+    private final Runnable task;
+    private final long delay;
+
+    public RunnableDelayPairing(Runnable runnable, long delay) {
+        this.task = runnable;
+        this.delay = delay;
+    }
+
+    public RunnableDelayPairing(Runnable runnable) {
+        this.task = runnable;
+        this.delay = 0;
+    }
+
+    public Runnable getTask() {
```

Comment:
This class is a good candidate to show you Lombok (you can see a short video about Lombok here https://projectlombok.org/)
You can add the following dependency to your `pom.xml` file

`<dependencies>
	<dependency>
		<groupId>org.projectlombok</groupId>
		<artifactId>lombok</artifactId>
		<version>1.18.20</version>
		<scope>provided</scope>
	</dependency>
</dependencies>`

Then, you can annotate the class with `@Getter` annotation. When you compile the code, in `target` directory you will see that compiled `RunnableDelayPairing` class has getter methods for `task` and `delay` fields.

Lombok is a tool to minimise the amount of boilerplate code in our classes.